dashboard/src/actions/overviewActions.js

```diff
@@ -290,17 +305,23 @@ export const editMetadata =
 
     const rIndex = data.findIndex((item) => item.resource_id === rId);
     data[rIndex][metadata] = value;
-    data[rIndex][CONSTANTS.IS_DIRTY] = true;
-    if (value.length > CONSTANTS.DATASET_NAME_LENGTH) {
-      data[rIndex][CONSTANTS.NAME_VALIDATED] = CONSTANTS.ERROR;
-      data[rIndex][
-        CONSTANTS.NAME_ERROR_MSG
-      ] = `Length should be < ${CONSTANTS.DATASET_NAME_LENGTH}`;
-    } else if (value.length === 0) {
-      data[rIndex][CONSTANTS.NAME_VALIDATED] = CONSTANTS.ERROR;
-      data[rIndex][CONSTANTS.NAME_ERROR_MSG] = `Length cannot be 0`;
+
+    if (metadata === "name") {
+      if (value.length > CONSTANTS.DATASET_NAME_LENGTH) {
+        data[rIndex][CONSTANTS.NAME_VALIDATED] = CONSTANTS.ERROR;
+        data[rIndex][
+          CONSTANTS.NAME_ERROR_MSG
+        ] = `Length should be < ${CONSTANTS.DATASET_NAME_LENGTH}`;
+      } else if (value.length === 0) {
+        data[rIndex][CONSTANTS.NAME_VALIDATED] = CONSTANTS.ERROR;
+        data[rIndex][CONSTANTS.NAME_ERROR_MSG] = `Length cannot be 0`;
+      } else {
+        data[rIndex][CONSTANTS.NAME_VALIDATED] = CONSTANTS.SUCCESS;
+      }
+      data[rIndex][CONSTANTS.IS_DIRTY][CONSTANTS.DATASET_NAME_KEY] = true;
     } else {
-      data[rIndex][CONSTANTS.NAME_VALIDATED] = CONSTANTS.SUCCESS;
+      data[rIndex][CONSTANTS.IS_DIRTY][CONSTANTS.SERVER_DELETION_KEY] = true;
+      data[rIndex].metadata[CONSTANTS.SERVER_DELETION] = value;
     }
```

Comment:
You should consider changing this `else` to an `else if`:  it's true, at the moment, that if `metadata` is not `"name"` then it must be `"server.deletion"`, but it would be worthwhile confirming that, just to protect against future expansions to the code.

Reply:
There's a Jira "wishlist" item to allow seeing and modifying arbitrary metadata values through the dashboard... so that would be consistent.

There are a few metadata items, specifically `dataset.name` and `server.deletion` which have special semantics ... the dashboard doesn't *have* to give them any special treatment so long as it handles a server validation error, but it's fine to check them here and just pass any other metadata key/value pairs through as-is.